src/main/java/jenkins/scm/api/SCMEvent.java

```diff
@@ -106,6 +107,9 @@
      * @since 2.0.3
      */
     public static final String ORIGIN_UNKNOWN = "?";
+
+    private static final int EVENT_THREAD_POOL_SIZE = SystemProperties
+        .getInteger(SCMEvent.class.getName() + ".EVENT_THREAD_POOL_SIZE", 30);
```

Comment:
debating whether or not to increase this or leave it at the previous value of 10, was also wondering about dynamically increasing the thread pool size if the queue get's too big.